dolphinscheduler-server/src/test/java/org/apache/dolphinscheduler/server/master/DependentTaskTest.java

```diff
@@ -37,188 +45,426 @@
-
+                .submitTask(Mockito.argThat(taskInstance -> taskInstance.getId() == 1000)))
+                .thenAnswer(i -> taskInstance);
 
-        Mockito.when(processService.findProcessInstanceById(10111))
-                .thenReturn(getProcessInstance());
-        Mockito.when(processService.findProcessDefineById(0))
-                .thenReturn(getProcessDefinition());
-        Mockito.when(processService.saveTaskInstance(getTaskInstance()))
+        // for DependentTaskExecThread.initTaskParameters
+        Mockito.when(processService
+                .updateTaskInstance(Mockito.any()))
+                .thenReturn(true);
+        // for DependentTaskExecThread.updateTaskState
+        Mockito.when(processService
+                .saveTaskInstance(Mockito.any()))
                 .thenReturn(true);
 
-        applicationContext = Mockito.mock(ApplicationContext.class);
-        SpringApplicationContext springApplicationContext = new SpringApplicationContext();
-        springApplicationContext.setApplicationContext(applicationContext);
-        Mockito.when(applicationContext.getBean(ProcessService.class)).thenReturn(processService);
-        Mockito.when(applicationContext.getBean(MasterConfig.class)).thenReturn(config);
+        // for DependentTaskExecThread.waitTaskQuit
+        Mockito.when(processService
+                .findTaskInstanceById(1000))
+                .thenAnswer(i -> taskInstance);
     }
 
+    /**
+     * task basic function for DependentTask
+     * TODO:
+     */
     @Test
-    public void testDependAll() throws Exception{
-
-        TaskInstance taskInstance = getTaskInstance();
-        String dependString = "{\"dependTaskList\":[{\"dependItemList\":[{\"dateValue\":\"today\",\"depTasks\":\"ALL\",\"projectId\":1,\"definitionList\":[{\"label\":\"C\",\"value\":4},{\"label\":\"B\",\"value\":3},{\"label\":\"A\",\"value\":2}],\"cycle\":\"day\",\"definitionId\":4}],\"relation\":\"AND\"}],\"relation\":\"AND\"}";
-        taskInstance.setDependency(dependString);
-
-        Mockito.when(processService.submitTask(taskInstance))
-                .thenReturn(taskInstance);
-        DependentTaskExecThread dependentTask =
-                new DependentTaskExecThread(taskInstance);
-
-        dependentTask.call();
-
-        Assert.assertEquals(ExecutionStatus.SUCCESS, dependentTask.getTaskInstance().getState());
-
-        DateInterval dateInterval =DependentDateUtils.getTodayInterval(new Date()).get(0);
-
-
-        Mockito.when(processService
-                .findLastRunningProcess(4, dateInterval.getStartTime(),
-                        dateInterval.getEndTime()))
-                .thenReturn(findLastStopProcessInterval());
-        DependentTaskExecThread dependentFailure = new DependentTaskExecThread(taskInstance);
-        dependentFailure.call();
-        Assert.assertEquals(ExecutionStatus.FAILURE, dependentFailure.getTaskInstance().getState());
+    public void testBasic() throws Exception {
```

Comment:
Some tests are full of NPE, please mock enough and right beans to avoid this situation.
Thx a lot~
![image](https://user-images.githubusercontent.com/29545877/89850451-f50fa800-dbbc-11ea-8b55-9da7940a5fb6.png)

Reply:
I think it is maybe a exception on MasterBaseTaskExecThread by design to warn about bad log config. But the only way to enable the log config is written in start script, which can be optimized (but not in this PR).